src/main/java/seedu/address/model/room/Room.java

```diff
@@ -123,18 +124,29 @@ public Room checkIn() {
     }
 
     /**
-     * Checks out the first booking of this room and its current occupant.
-     * Future features to include exporting of receipt, setting room to housekeeping status for __x__ hours.
+     * Checks out the first booking of this room.
+     * KIV: Future features to include exporting of receipt, setting room to housekeeping status for __x__ hours.
      */
     public Room checkout() {
-        if (!hasActiveOrExpiredBooking()) {
-            throw new NoActiveOrExpiredBookingException();
-        }
         Booking firstBooking = bookings.getFirstBooking();
         return makeRoom(this.roomNumber, this.expenses, bookings.remove(firstBooking), this.tags);
         // expenses.report(); // weizheng implement this later
     }
 
+    /**
+     * Checks out the booking identified by the booking period
+     * KIV: Future features to include exporting of receipt, setting room to housekeeping status for __x__ hours.
+     */
+    public Room checkout(BookingPeriod bookingPeriod) {
+        for (Booking booking : bookings.getSortedBookingsSet()) {
+            if (booking.getBookingPeriod().equals(bookingPeriod)) {
+                return makeRoom(this.roomNumber, this.expenses, bookings.remove(booking), this.tags);
+            }
+        }
+        throw new BookingNotFoundException();
+        // expenses.report(); // weizheng implement this later
```

Comment:
Could write this "to implement" part in the javadoc instead. Similar to the KIV section.

Reply:
agree